branchLayout/api/src/test/java/com/virtuslab/branchlayout/api/BranchLayoutTestSuite.java

```diff
@@ -127,6 +127,121 @@ public void withBranchSlideOut_givenNonRootExistingBranch_slidesOut() {
     assertEquals(rootName, children.get(1).getParent().getName());
   }
 
+  @Test
+  public void withBranchRename_givenBranchLayout_renamesChild() {
+    // given
+    String rootName = "root";
+    String branchToRename = "child";
+    String newBranchName = "kinder";
+
+    /*-
+        root         rename         root
+          child      ----->           kinder
+    */
+
+    List<BranchLayoutEntry> childBranches = List.of(
+        new BranchLayoutEntry(branchToRename, /* customAnnotation */ null, List.empty()));
+
+    val rootEntry = new BranchLayoutEntry(rootName, /* customAnnotation */ null, childBranches);
+    val branchLayout = new BranchLayout(List.of(rootEntry));
+
+    // when
+    BranchLayout result = branchLayout.rename(branchToRename, newBranchName);
+
+    // then
+    assertEquals(result.getRootEntries().size(), 1);
```

Comment:
Nit: keep expected than actual order in newly-added tests (unless that's already fixed on JUnit PR)

Reply:
it will be fixed in JUnit PR - I'd rather keep it whatever it is here